pam/internal/adapter/model.go

```diff
@@ -257,6 +257,29 @@ func (m *UIModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			m.gdmModel, gdmCmd = m.gdmModel.Update(msg)
 		}
 
+		// If the mode is passwd (meaning we are trying to change the user password) and we already have a AUTHTOK, we
+		// don't need to ask the user for a new one.
+		if msg.layout.Type == "newpassword" && m.SessionMode == authd.SessionMode_PASSWD {
+			challenge, err := m.PamMTx.GetItem(pam.Authtok)
+			if err == nil && challenge != "" {
```

Comment:
Maybe handle the error case? Since it's a fatal failure if we have one here.

Reply:
It's not a fatal failure. If we don't find any authtok, we can proceed with the previous behavior of asking the user for the new password (which means we will also support PAM configurations that do not have pwquality enabled).

Reply:
No, sure `challenge == ""` is not a fatal error, but if `err` is returned it is, since it means something is broken in the pam side or a programmer erorr, so I don't think it should be ignored.